nilearn/maskers/nifti_labels_masker.py

```diff
@@ -553,25 +580,38 @@ def fit(
         logger.log(msg=msg, verbose=self.verbose)
         self.labels_img_ = _utils.check_niimg_3d(self.labels_img)
 
-        # create _region_id_name dictionary
-        # this dictionary will be used to store region names and
-        # the corresponding region ids as keys
-        self._region_id_name = None
-        if self.labels is not None:
-            known_backgrounds = {"background", "Background"}
-            initial_region_ids = [
+        # find the region ids existing in the self.labels_img_
+        initial_region_ids = [
+            region_id
+            for region_id in np.unique(
+                _utils.niimg.safe_get_data(self.labels_img_)
+            )
+            if region_id != self.background_label
+        ]
+        if self._region_id_name is not None:
+            # check if initial_region_ids all exist in self._region_id_name
```

Comment:
here I would do 2 checks:
- see if we have regions in the tsv that do not exist in the image
- see if we have regions in the image that do not exist in the tsv

I wonder if in the long term we could just throw warnings for those an not errors.

Reply:
I think the first one is not critical, but it can be a warning.